public/app/features/logs/components/log-context/LogRowContextModal.tsx

```diff
@@ -102,27 +104,54 @@ const getStyles = (theme: GrafanaTheme2) => {
         color: ${theme.colors.text.link};
       }
     `,
+    loadingCell: css`
+      position: sticky;
+      left: 50%;
+      display: inline-block;
+      transform: translateX(-50%);
+    `,
   };
 };
 
-export enum LogGroupPosition {
-  Bottom = 'bottom',
-  Top = 'top',
-}
-
 interface LogRowContextModalProps {
   row: LogRowModel;
   open: boolean;
   timeZone: TimeZone;
   onClose: () => void;
-  getRowContext: (row: LogRowModel, options?: LogRowContextOptions) => Promise<DataQueryResponse>;
+  getRowContext: (row: LogRowModel, options: LogRowContextOptions) => Promise<DataQueryResponse>;
 
   getRowContextQuery?: (row: LogRowModel, options?: LogRowContextOptions) => Promise<DataQuery | null>;
-  logsSortOrder?: LogsSortOrder | null;
+  logsSortOrder: LogsSortOrder;
   runContextQuery?: () => void;
   getLogRowContextUi?: DataSourceWithLogsContextSupport['getLogRowContextUi'];
 }
 
+type Place = 'above' | 'below';
```

Comment:
What do you think about changing this to an enum?

Reply:
Well, it's not exported, but should be :D [LoadingIndicator.tsx](https://github.com/grafana/grafana/pull/69459/files#diff-4003b3552c8c5c6184de49c8aff44be9bb2c78c0ba2306fce3ac9560a69d5fb9) is using that type as well. But I think that's fine for now

Reply:
thanks all, good point about `LoadingIndicator`, i fixed that.